docs/status.md

```diff
@@ -1,52 +1,109 @@
+
+Additionally, we define 3 levels of maturity: **alpha**, **beta**, and **generally available (GA)**. Each of these maturity levels will be combined with a threshold of the factors above. When a language meets the maturity threshold for each of the factors then it’s moved into that maturity level.
+
+The following thresholds define each maturity level:
+
+* **Alpha**
+    * Parse rate: 90%+
+    * Rulesets: 0+ (10+ rules)
+    * Features:
+        * `concrete_syntax`
+        * `deep_exprstmt`
+        * `dots_args`
+        * `dots_nested_stmts`
+        * `dots_stmts`
+        * `dots_string`
+        * `metavar_arg`
+        * `metavar_call`
+        * `metavar_equality_var`
+* **Beta**
+    * Parse rate: 99%+
+    * Rulesets: 1+ (10+ rules)
+    * Features:
+        * All in alpha
+        * `class_def`
+        * `func_def`
+        * `metavar_cond`
+        * `metavar_equality_expr`
+        * `metavar_equality_stmt`
+        * `metavar_import`
+        * `metavar_stmt`
+* **Generally Available (GA)**
+    * Parse rate: 99.9%+
+    * Rulesets: 2+ (10+ rules)
+    * Features:
+        * All in alpha
+        * All in beta
+        * `annotations` (language dependent)
+        * `deep_expr_operator`
+        * `equivalence_constant_propagation`
+        * `equivalence_eq`
+        * `equivalence_naming_import` (language dependent)
+        * `key_value`
+        * `metavar_typed` (language dependent)
+        * `regexp_string`
+
+Generally speaking, the features-by-maturity level are roughly: **alpha)** syntax support, ellipsis operator support, and basic metavariable support, **beta)** everything prior and nearly complete metavariable support and metavariable equality, **GA)** everything prior and all exotic features like regexp, equivalence, deep expression operator, typing, etc.
+
+Visit the cheatsheet generation script and associated semgrep-core test files to learn more about each feature:
+
+* [https://github.com/returntocorp/semgrep/blob/develop/scripts/generate_test_matrix.py](https://github.com/returntocorp/semgrep/blob/develop/scripts/generate_test_matrix.py)
+* [https://github.com/returntocorp/semgrep/tree/develop/semgrep-core/tests](https://github.com/returntocorp/semgrep/tree/develop/semgrep-core/tests)
+
+## Support Expectations
+
+* **Alpha**: experimental support with many known bugs.
+    * Looking for dedicated users to help us improve these languages.
+    * Expect limited support responses, as these languages will be lowest priority.
+* **Beta**: supported language with known bugs.
+    * Looking for beta users to report bugs and rapidly iterate with our team.
+    * Expect best-effort support responses when there are no higher priority requests being handled.
+* **GA**: production-level support with few known bugs.
+    * Looking for bug reports and feedback from users.
+    * Expect timely and thorough support responses, generally within 24 hours.
+
+## Language Parse Rate
+
+<table>
```

Comment:
I think the table looks nice on desktop but we lose the re-flow of blocks on mobile compared to the sequence of divs we had previously, which causes the other content to be squeezed into a narrower container:

<img width="425" alt="Screen Shot 2020-12-03 at 9 07 23 AM" src="https://user-images.githubusercontent.com/302941/101062933-1408c000-3547-11eb-9a9d-8893ec0fd814.png">

Reply:
Hmm, that makes sense to me. I originally went this route because adding markdown to the existing HTML format was giving me weird formatting issues. Not sure how best to proceed here.

I was hoping to eventually automate this and include feature checklist, parse rate, and rulesets for each language at its current maturity level. Somewhat tangentially, this takes up a lot of space just to provide a number for each language :man_shrugging:

Reply:
What weird issues did you get? 

Previously we had it in this form:
```
<div class="lang-container">
  <h2>Go parse rate</h2>
  <iframe width="450" height="150" frameBorder="0" src="https://dashboard.semgrep.dev/metric/semgrep.core.go.parse.pct/number"></iframe>
</div>

<div class="lang-container">
  <h2>Java parse rate</h2>
  <iframe width="450" height="150" frameBorder="0" src="https://dashboard.semgrep.dev/metric/semgrep.core.java.parse.pct/number"></iframe>
</div>

...
```
and there's a tiny bit of styling applied to `.lang-container` just to give it some breathing room.

Agree about the space required to show a number. I think this was about as small as I could get it (after moving from a graph iframe to a number iframe), but I can't remember how much I tried to play with the iframe sizing to squeeze it. One downside of the iframe is that you can't target CSS for elements inside the frame.

Reply:
Here's what I see when I include the new content on the original formatting:

![image](https://user-images.githubusercontent.com/1724818/101085121-382ac800-356c-11eb-80c5-c87b19456d35.png)

Reply:
> Here's what I see when I include the new content on the original formatting:
> 

Hmm yeah that does look odd. When I pull your branch and replace the table with the current list of divs on that page, it looks OK on Chrome, FF, and Safari. Are you using something different?

Reply:
Are you including this part?

```
---
template: stats-template.html
title: Semgrep Supported Languages
---
```

Without that I see:

![image](https://user-images.githubusercontent.com/1724818/101088816-74acf280-3571-11eb-9f01-9de31c3f632f.png)

With that I see:

![image](https://user-images.githubusercontent.com/1724818/101088851-855d6880-3571-11eb-9a47-32930d458006.png)

This is Firefox on Linux, not sure if that's affecting things. Want to try pushing your changes up to this PR? I'm not sure I'm fully replicating what you have.

Reply:
Sure I can do that, one sec.